plugins/ezid/logic.py

```diff
@@ -196,45 +182,36 @@ def update_doi_via_ezid(ezid_config, ezid_metadata):
     # ezid_config dictionary contains values for the following keys: shoulder, username, password, endpoint_url
     # ezid_metadata dicitionary contains values for the following keys: update_id, target_url, group_title, contributors, title, published_date, accepted_date
 
-    # pdb.set_trace()
-
-    posted_content = {
-        "posted_content": {
-            "@xmlns": "http://www.crossref.org/schema/4.4.0",
-            "@xmlns:xsi": "http://www.w3.org/2001/XMLSchema-instance",
-            "@xmlns:jats": "http://www.ncbi.nlm.nih.gov/JATS1",
-            "@xsi:schemaLocation": "http://www.crossref.org/schema/4.4.0 http://www.crossref.org/schema/deposit/crossref4.4.0.xsd",
-            "@type": 'preprint',
-            "group_title": ezid_metadata['group_title'],
-            "contributors": ezid_metadata['contributors'],
-            "titles": {
-                "title": ezid_metadata['title']
-            },
-            "posted_date": ezid_metadata['published_date'],
-            "acceptance_date": ezid_metadata['accepted_date'],
-            "doi_data": {"doi": ezid_metadata['update_id'], "resource": ezid_metadata['target_url']}
-        }
-    }
-
-    metadata = unparse(posted_content).replace('\n', '').replace('\r', '')
+    if ezid_metadata.get('published_doi') is not None:
+        #we cannot trust that the published_doi has been validated, or is usable as a URL, so let's do that now
+        logger.debug('validating published_doi')
+        validator = URLValidator()
+        try:
+            validator(ezid_metadata.get('published_doi'))
+        except ValidationError:
+            logger.error('invalid URL, published_doi: %s for preprint: %s', ezid_metadata.get('published_doi'), ezid_metadata.get('target_url'))
```

Comment:
Hmm... this should probably drop the published_doi, since it's invalid, instead of jovially continuing after logging the error